source/02_algo/distributed_rl_eng.rst

```diff
@@ -0,0 +1,115 @@
+
+
+Research Direction
+---------
+
+System
+~~~~~~
+
+Overall Architecture
+^^^^^^^^^^
+For common decision problems, the two most commonly used distributed architectures are IMPALA [3]_ and SEED RL [4]_
+
+.. image:: ./images/impala.png
+  :align: center
+  
+- The former is the classic Actor-Learner mode; that is, the data collection and training sides are entirely separated, and the latest neural network model is regularly passed from the Learner to the Actor, and the Actor is sent to the Learner after collecting a certain amount of data (i.e. observations). If there are multiple Learners, they also periodically synchronize the gradients of the neural network (i.e. the data-parallel model in distributed deep learning).
+
+.. image:: ./images/seed_rl.png
+  :scale: 50%
+  :align: center
+
+- On the basis of the former, the latter is dedicated to optimizing the loss of the transmission model. SEED RL strips out the part used for inference to generate actions, and puts it together with the training end to update the model through efficient inter-TPU communication technology. The cost of passing the model in IMPALA is greatly reduced, and for cross-machine communication between the environment and the reasoning Actor, SEED RL uses an optimized gRPC scheme to pass the observation and action, so there is not much burden.
+
+.. note::
+There is no absolute superiority or inferiority between these two schemes. The key lies in the fact that for a practical decision-making problem, whether it is more expensive to transmit models across machines, or more expensive to transmit observation and action data across machines, if it is the former , and there are better communication components between GPU/TPU, then SEED RL is a better solution, if it is the latter, IMPALA is a more stable choice. In addition, IMPALA can accumulate a batch of data for data transmission, while SEED RL requires data transmission in each interactive frame. This is a classic data batch and stream processing comparison problem. For the current machine learning community, the former is generally more complex and user-friendly. Also, if the entire training procedure requires a higher degree of freedom and customization, such as dynamically controlling some behavior of the Actor, IMPALA is more convenient.
+
+In addition to the above two architectures, there are many other distributed reinforcement learning design schemes, such as A3C [5]_ and Gossip A2C [6]_ that introduce asynchronous neural network update schemes, In order to support large-scale self-play, AlphaStar [1]_ with a complex League mechanism was designed, and MuZero [7]_ combined with model-based RL and MCTS-related modules will not be described here. Interested readers can refer to the specific Papers or refer to our `Algorithm Raiders Collection section <../12_policies/index_zh.html>`_.
```

Comment:
- change: schemes. In order
- i think it is not necessary to specify: will not be described here

Reply:
thanks for the comment. Which word should I change "schemes" to?

Reply:
Sorry, i was not clear here, schemes is correct, but there is a small typos error ", In order" change to ". In order", just change comma with dot

Reply:
done!